lib/intercept.js

```diff
@@ -430,13 +422,7 @@ function activate() {
 
       //  NOTE: Since we already overrode the http.ClientRequest we are in fact constructing
       //    our own OverriddenClientRequest.
-      req = new http.ClientRequest(options)
-
-      res = RequestOverrider(req, options, interceptors, remove)
```

Comment:
Am I right that this was removed because it shoudn't have been here in the first place ≠ because this work is already done inside the `http.ClientRequest` constructor?

Reply:
This call was already doing nothing except getting a ref to the response so it could add the callback listener. But since the `ClientRequest` accepts a callback, it's a lot cleaner to pass it through and not have the duplicate logic.

Reply:
That part makes sense, and I agree it's much cleaner.

What I'm confused about is that I don't see where you're creating the new response, to replace the one being removed here.

Reply:
The var `res` was poorly named. It was never a response instance. If you look at the [bottom of the `RequestOverrider` function](https://github.com/nock/nock/blob/beta/lib/request_overrider.js#L570), it just returns the request instance passed in as the first arg.

So the old code was effectively:
```js
req = new http.ClientRequest(options)

// res = RequestOverrider(req, options, interceptors, remove)

if (callback) {
  // res.on('response', callback)
  req.on('response', callback)
}
return req
```

I assume in the past, calling `RequestOverrider` had some other purpose, but it became dead code.
Keeping in mind that `RequestOverrider` is also called with the req and options in `OverriddenClientRequest`.

Reply:
D'oh! Thanks for explaining that; I appreciate it.

This construction and override logic is so obfuscated! I really look forward to some serious cleanup in here.